tests/CSharp/CSharp.h

```diff
@@ -1487,7 +1487,8 @@ struct TestVariableWithoutType
 class DLL_API StringMarshall
 {
 public:
-    static constexpr const char* CSharpString = "C#";
-    static bool CSharpString8(const char* in);
-    static bool CSharpString16(const char16_t* in);
+    static const char* TestCSharpString(const char* in, CS_OUT const char** out);
```

Comment:
I've been wondering why your tests are so complicated. You could write them in the way I have: [you pass a parameter and then return the parameter](https://github.com/mono/CppSharp/blob/master/tests/Common/Common.h#L1527). This way you easily and fully test both directions.
Also, you don't need a class for this particular test, free functions are enough.